server/utils/retrieveQuestionResponseFromApplicationOrAssessment.ts

```diff
@@ -0,0 +1,59 @@
+import { ApprovedPremisesApplication } from '@approved-premises/api'
+import { TasklistPageInterface } from '../form-pages/tasklistPage'
+import { getPageDataFromApplicationOrAssessment, getPageName } from '../form-pages/utils'
+import { SessionDataError } from './errors'
+import { camelCase } from './utils'
+
+/**
+ * Retrieves response for a given question from the application object or throws an error if it does not exist.
+ * @param application the application to fetch the response from.
+ * @param task the task to retrieve the response for.
+ * @param page the page that we need the response for in camelCase.
+ * @param {string} question [question=page] the page that we need the response for. Defaults to the value of `page`.
+ * @returns the response for the given task/page/question.
+ */
+export const retrieveQuestionResponseFromApplicationOrAssessment = <T>(
+  applicationOrAssessment: ApprovedPremisesApplication,
+  Page: unknown,
+  question?: string,
+) => {
+  const pageData = getPageDataFromApplicationOrAssessment(Page as TasklistPageInterface, applicationOrAssessment)
```

Comment:
Yeah, some of the pages don't match the `TasklistPageInterface` for some reason. I thought it'd be better to just cast rather than getting stuck in the weeds.